### PR TITLE
21104 - add business search validation

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#21104

*Description of changes:*

- Added validation check for EXPRO business making a call to Colin API

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py [hunk 1]
```diff
@@ -165,3 +169,22 @@ def validate_continuation_in_court_order(filing: dict, filing_type) -> list:
...
+    business_identifier_path = f'/filing/{filing_type}/business/identifier'
+
+    if filing_json['filing'][filing_type].get('business'):
+        identifier = filing_json['filing'][filing_type]['business']['identifier']
+        url = f'{current_app.config["COLIN_URL"]}/businesses/{identifier}/public'
+        headers = {
+            'Authorization': f'Bearer {jwt.get_token_auth_header()}',
+            'Content-Type': 'application/json'
+        }
+        response = requests.get(url, headers=headers)
```
Comment (ketaki-deodhar) [inline legal-api/src/legal_api/services/filings/validations/continuation_in.py:186]: Do we need to check anything else other than Business was found?

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py [hunk 2]
```diff
@@ -165,3 +169,22 @@ def validate_continuation_in_court_order(filing: dict, filing_type) -> list:
...
+        if response.status_code == HTTPStatus.NOT_FOUND:
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/filings/validations/continuation_in.py:187]:
```suggestion
        if response.status_code != HTTPStatus.OK:
```

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/continuation_in.py:186]: may be you can compare the legal name (https://github.com/bcgov/lear/blob/main/queue_services/entity-filer/src/entity_filer/filing_processors/continuation_in.py#L52)

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/continuation_in.py:186]: Can you create a new service `colin.py` and move this api call to services folder (similar to namex)

File: legal-api/src/legal_api/services/colin.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class ColinService():
+    """Provides services to use the colin-api."""
+
+    @staticmethod
+    def query_business(identifier: str):
+        """Return a JSON object with business information."""
+        # Perform proxy call with identifier
+        url = f'{current_app.config["COLIN_URL"]}/businesses/{identifier}/public'
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/colin.py:29]:
```suggestion
        url = f'{current_app.config["COLIN_URL"]}/api/v1/businesses/{identifier}/public'
```
image

File: legal-api/src/legal_api/services/colin.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Copyright © 2020 Province of British Columbia
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/services/colin.py:1]: 2024

**Review by vysakh-menon-aot — APPROVED** [review]

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py [hunk 5]
```diff
@@ -165,3 +167,23 @@ def validate_continuation_in_court_order(filing: dict, filing_type) -> list:
...
+        legal_name = filing_json['filing'][filing_type]['business']['legalName']
+        response = colin.query_business(identifier)
+        response_json = response.json()
+        if response.status_code != HTTPStatus.OK:
+            msg.append({'error': 'Could not fetch business data for company from Colin.',
+                        'path': business_identifier_path})
+        elif legal_name != response_json['business']['legalName']:
+            msg.append({'error': 'Legal name does not match with company legal name from Colin.',
+                        'path': business_legal_name_path})
+    return msg
```
Comment (severinbeauvais) [inline legal-api/src/legal_api/services/filings/validations/continuation_in.py:189]: Yeah, I guess you don't have to verify much here, only that the business exists.

Here's what the UI code does with the COLIN business data: https://github.com/bcgov/business-create-ui/blob/508671b0aaac1cad25ff1c6b253a8bf283cbc4ea/src/components/ContinuationIn/ExtraproRegistration.vue#L531

**Review by severinbeauvais — APPROVED** [review]

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py (hunk 5 above)
Comment (ketaki-deodhar) [reply legal-api/src/legal_api/services/filings/validations/continuation_in.py:189]: Yes

**Pull request merged**